### ISSUE TITLE
[iOS] Crash login

### ISSUE THREAD

**Issue by pboulch** [NativeScript/nativescript-facebook#184]
### Which platform(s) does your issue occur on?
* iOS
* device
### Please, provide the following version numbers that your issue occurs with:
* Cross-platform modules: 5.2.0
* Runtime(s): tns-ios: 5.2.0

### Please, tell us how to recreate the issue in as much detail as possible.
On iOS devices, when I try to login (on click into an Angular component) :
```
ERROR Error: *** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[3]
```

I followed the instructions to install the plugin : 
* Add lines into Info.plist
* Initialize the plugin into the main.ts (with Facebook AppID)

Let me know if further information is needed.

**Comment by jannomeister** [comment]
+1 also experiencing this issue.

**Comment by pboulch** [comment]
Finally, I found a fix, as I use the firebase plugin, I activated Facebook on the firebase plugin and the plugin Facebook (this one) now works

**Comment by gogoout** [comment]
Seems the problem only occured in `4.44`, I generated the types and found no changes in there, therefore I was not able to fix the issue.
So I guess the current workround is manually edit the version in podfile to below.

```podfile
pod 'FBSDKCoreKit', '~> 4.40.0'
pod 'FBSDKLoginKit', '~> 4.40.0'
pod 'FBSDKShareKit', '~> 4.40.0'
```

**Comment by fthuin** [comment]
> Finally, I found a fix, as I use the firebase plugin, I activated Facebook on the firebase plugin and the plugin Facebook (this one) now works. But it's only a temporary fix. Maybe the problem is the Facebook SDK version ... I don't know

If you do that, you'll have duplicate dependencies on `FBSDKCoreKit` and `FBSDKLoginKit` which is not a very good practice.

@DimitarTodorov Would you accept a PR with a more specific Pod version as proposed by @gogoout ?

**Comment by gogoout** [comment]
@fthuin that's just a temporary workaround, not a fix. I can't find any active bug related to this one from Facebook's official bug tracker. So someone has to looking into it, otherwise the problem will remain in the later sdk.

**Comment by fthuin** [comment]
> @fthuin that's just a temporary workaround, not a fix. I can't find any active bug related to this one from Facebook's official bug tracker. So someone has to looking into it, otherwise the problem will remain in the later sdk.

I agree it's a temporary fix, I don't think it's a "bug" in the facebook SDK, it's just the typings generated in this plugin are not coherent with the latest version of the Facebook SDK anymore (because they deprecated some stuff we use https://github.com/facebook/facebook-objc-sdk/commit/6c4ac12f42916c7e104469002973747a2e5c4c83 ), so I think a minor release with a more specific Pod version is a quick fix; the long-term fix is to regenerate typings and use the new function `logInWithPermissions` instead of the `logInWithReadPermissions` which was deprecated.

**Comment by VladimirAmiorkov** [comment]
Hi @fthuin @gogoout ,

We are investigating this issue and any additional finding on your part would be much appreciated. Any PRs are welcome as always.

**Comment by VladimirAmiorkov** [comment]
Hi @fthuin @gogoout ,

You can follow these two PRs which both resolve this issue:
https://github.com/NativeScript/nativescript-facebook/pull/188
https://github.com/NativeScript/nativescript-facebook/pull/189

We are planning to merge the first one and release a patch version and after merging the second one you can look forward to a major version.

**Comment by DimitarTodorov** [comment]
https://github.com/NativeScript/nativescript-facebook/releases/tag/v3.0.1
Fixed with 3.0.1 of nativescript-facebook
